src/Octoshift/GithubClient.cs

```diff
@@ -170,13 +170,14 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url, Dictionary
             {
                 (content, headers) = await SendAsync(httpMethod, url, body, expectedStatus, customHeaders);
             }
-            else if (expectedStatus == HttpStatusCode.OK)
+            else if (response.StatusCode == HttpStatusCode.OK)
             {
                 response.EnsureSuccessStatusCode();
             }
             else if (response.StatusCode != expectedStatus)
             {
-                throw new HttpRequestException($"Expected status code {expectedStatus} but got {response.StatusCode}", null, response.StatusCode);
+
```

Comment:
We never got to this code until now as EnsureSuccessStatusCode was throwing an HttpRequestException for non-200 responses. Made it where we are now more explicit in the message (since the status code is already part of the exception) which gives us the opportunity to catch specific exceptions like I am doing in the API.